javascript/src/components/streams/StreamComponent.jsx

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+var React = require('react/addons');
+var StreamsStore = require('../../stores/streams/StreamsStore');
+var StreamList = require('./StreamList');
+var CreateStreamButton = require('./CreateStreamButton');
+var SupportLink = require('../support/SupportLink');
+var PermissionsMixin = require('../../util/PermissionsMixin');
+var StreamRulesStore = require('../../stores/streams/StreamRulesStore');
+
+var StreamComponent = React.createClass({
+    mixins: [PermissionsMixin],
+    getInitialState() {
+        return {
+            streams: []
+        };
+    },
+    componentDidMount() {
+        this.loadData();
+        StreamsStore.onChange(this.loadData);
+        StreamRulesStore.onChange(this.loadData);
```

Comment:
We only load the stream rule types in this component from this store, but the idea was that the component is rerendering when the set of stream rules changes.